var/spack/repos/builtin/packages/nalu/package.py

```diff
@@ -31,21 +31,21 @@ class Nalu(CMakePackage):
 
     # Required dependencies
     depends_on("mpi")
-    depends_on("yaml-cpp@0.5.3:", when="+shared")
-    depends_on("yaml-cpp~shared@0.5.3:", when="~shared")
+    depends_on("yaml-cpp@0.5.3:0.6.2", when="+shared")
+    depends_on("yaml-cpp~shared@0.5.3:0.6.2", when="~shared")
     # Cannot build Trilinos as a shared library with STK on Darwin
     # which is why we have a 'shared' variant for Nalu
     # https://github.com/trilinos/Trilinos/issues/2994
     depends_on(
-        "trilinos"
+        "trilinos@develop"
```

Comment:
```suggestion
        "trilinos@master:"
```
You can use the versioning semantics to allow for multiple named versions. See bullet 1 in 
https://spack.readthedocs.io/en/latest/packaging_guide.html#version-comparison